src/amr_criteria/refinement_package.cpp

```diff
@@ -20,19 +20,25 @@
 #include <utility>
 
 #include "amr_criteria/amr_criteria.hpp"
+#include "interface/make_pack_descriptor.hpp"
 #include "interface/mesh_data.hpp"
 #include "interface/meshblock_data.hpp"
 #include "interface/state_descriptor.hpp"
+#include "kokkos_abstraction.hpp"
 #include "mesh/mesh.hpp"
 #include "mesh/mesh_refinement.hpp"
 #include "mesh/meshblock.hpp"
 #include "parameter_input.hpp"
+#include "utils/instrument.hpp"
 
 namespace parthenon {
 namespace Refinement {
 
 std::shared_ptr<StateDescriptor> Initialize(ParameterInput *pin) {
   auto ref = std::make_shared<StateDescriptor>("Refinement");
+  bool check_refine_mesh =
+      pin->GetOrAddBoolean("parthenon/mesh", "CheckRefineMesh", true);
+  ref->AddParam("check_refine_mesh", check_refine_mesh);
```

Comment:
Ah, I now see that `parthenon/mesh/CheckRefineMesh` is also being used outside of the example.
Personally, I'm not a super big fan of these switches for callback functions (for most other callback functions we only allow to enroll the `MeshData` version or the `MeshBlockData` and check if both are enrolled [and then throw an error]).
I'd be in favor of removing the input file switch and go with that pattern, but more people should weigh in here.

Reply:
The main use is for deciding which `amr_criteria` variant to use, rather than for the callbacks.  Maybe there is a better name to avoid the confusion, something like `MeshDataCriteria`. I was avoiding removing the `MeshBlockData` derivative criteria, but it shouldn't be breaking.

As it is written nothing checks whether both callbacks are enrolled, but I can add that as you've described. I only used the switch to easily check that it was giving the same refinement patterns.

Reply:
I'd be nice to get additional input here (@lroberts36 @jdolence @bprather @brryan @Yurlungur ) with regard to whether this a pattern we generally want to introduce to the codebase.
I'm in favor of (potentially) breaking backwards compatibility (though I think it should also work without breaking) over introduce a runtime switch that determines the callback function (at the Parthenon level).

Reply:
I would vote for not introducing a runtime parameter that switches the callback function. I think we should be moving to using `MeshData` based functions everywhere and encouraging downstream users to do this as well, so I would also support removing the `MeshBlockData` based criteria as well (unless this ends up putting a significant burden on some downstream codes?).

Reply:
Entirely removing the `MeshBlockData` callback would be breaking. In riot at least that would require some effort, though not a lot, to conform. I think that would be better in the long term though and it wouldn't be too big of a burden, assuming `MeshData` is alwasy faster.

Reply:
The runtime switch isn't for the callbacks, but rather for the refinement criteria. The callbacks are treated the same as the `EstimateTimeStep` callbacks.

If we don't have the runtime switch then I think the `MeshBlockData` based amr_criteria would need to be removed. Unless there is some other way to keep both but not call both in each cycle.